### PR TITLE
[refactor-prompt] Migrate command: wallet delete_addr

### PR DESCRIPTION
**Pull request by LysanderGG**
**What current issue(s) does this address, or what feature is it adding?**
Migrate the command `wallet delete_addr` to the new Command system.

**How did you solve this problem?**
Implementation of `CommandWalletDeleteAddress`

**How did you make sure your solution works?**
Added unit tests & manual testing

**Are there any special changes in the code that we should be aware of?**
Not a change but `UserWallet.DeleteAddress` always returns True so even when using `wallet delete_addr unknown_address`, the prompt output is `Deleted address unknown_address`.
If we want to change this behaviour, let's discuss the details.

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [ ] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: neo/Prompt/Commands/tests/test_wallet_commands.py [hunk 1]
```diff
@@ -240,6 +240,32 @@ def test_wallet_create_address(self):
...
+
+        # test wallet delete address with no argument
+        args = ['delete_addr']
+        res = CommandWallet().execute(args)
+        self.assertFalse(res)
+
+        # test wallet delete address with unknown address
+        args = ['delete_addr', self.watch_addr_str]
+        res = CommandWallet().execute(args)
+        self.assertTrue(res)  # Current specs.
```
Comment (ixje) [inline neo/Prompt/Commands/tests/test_wallet_commands.py]: So as you noticed `DeleteAddress` _always_ returns `True`. Which can give confusing user feedback like the following for an address that's not actually in the wallet:
```
neo> wallet delete_addr AJN2SZJuF7j4mvKaMYAY9N8KsyD4j1fNdf
Deleted address AJN2SZJuF7j4mvKaMYAY9N8KsyD4j1fNdf
```

It's not really related to the migration but how the underlying `DeleteAdress` works. Would you like to fix it or shall I make it a separate issue?

Comment (LysanderGG) [reply neo/Prompt/Commands/tests/test_wallet_commands.py]: It's a small fix so I'll do it in this PR!

**Comment by LysanderGG** [comment]
@ixje 
5078100 handles your comment here https://github.com/CityOfZion/neo-python/pull/752#discussion_r242148127

I decided to return `success` returned by `Wallet.DeleteAddress`, please tell me if this is not correct.

I also made sure that (invalid) too short or too long addresses do not result in an exception dumped to the prompt but a user friendly message.

**Comment by ixje** [comment]
all good 👍

**Pull request merged**